kubejs/server_scripts/deathofinfinity.js

```diff
@@ -0,0 +1,117 @@
+            event.recipes.createFilling("minecraft:medium_amethyst_bud", ["minecraft:medium_amethyst_bud",{ fluid: "kubejs:shimmer", amount: 10 * mB }]),
+            event.recipes.createFilling("minecraft:medium_amethyst_bud", ["minecraft:medium_amethyst_bud",{ fluid: "minecraft:water", amount: BUCKET / 4 }]),
+            event.recipes.createFilling("minecraft:medium_amethyst_bud", ["minecraft:medium_amethyst_bud",{ fluid: "kubejs:shimmer", amount: 10 * mB }]),
+        ])
+        .transitionalItem("minecraft:medium_amethyst_bud")
+        .loops(1);
+
+        event.recipes
+        .createSequencedAssembly(
+            ["minecraft:amethyst_cluster"],
+            ["minecraft:large_amethyst_bud"],
+            [event.recipes.createFilling("minecraft:large_amethyst_bud", ["minecraft:large_amethyst_bud",{ fluid: "minecraft:water", amount: BUCKET / 4 }]),
+            event.recipes.createFilling("minecraft:large_amethyst_bud", ["minecraft:large_amethyst_bud",{ fluid: "kubejs:shimmer", amount: 10 * mB }]),
+            event.recipes.createFilling("minecraft:large_amethyst_bud", ["minecraft:large_amethyst_bud",{ fluid: "minecraft:water", amount: BUCKET / 4 }]),
+            event.recipes.createFilling("minecraft:large_amethyst_bud", ["minecraft:large_amethyst_bud",{ fluid: "kubejs:shimmer", amount: 10 * mB }]),
+        ])
+        .transitionalItem("minecraft:large_amethyst_bud")
+        .loops(1);
+
+const mixings = [
+    {
+        output: {fluid: "minecraft:lava", amount: 125 * mB },
+        input: [
+            "1x createastral:stone_dust",
+            "1x minecraft:cobblestone",
+            { fluid: "minecraft:lava", amount: 50 * mB },
+        ],
+        heat: "",
+        time: null,
+    },{
+    output: {fluid: "minecraft:lava", amount: 250 * mB },
+    input: [
+        "4x createastral:stone_dust",
+        "2x minecraft:cobblestone",
+        { fluid: "minecraft:lava", amount: 50 * mB },
+    ],
+    heat: "heated",
+    time: null,
+    },{
+        output: {fluid: "minecraft:lava", amount: 500 * mB },
+        input: [
+            "4x createastral:stone_dust",
+            "2x minecraft:netherack",
+        ],
+        heat: "superheated",
+        time: null,
+    },{
+        output: {fluid: "kubejs:shimmer", amount: 250 * mB },
+        input: [
+            "5x minecraft:amethyst_shard",
+            "2x minecraft:glow_ink_sac",
+        ],
+        heat: "",
+        time: null,
+    },{
+    output: {fluid: "kubejs:shimmer", amount: 500 * mB },
+    input: [
+        "5x techreborn:amethyst_dust",
+        "2x minecraft:glow_ink_sac",
+    ],
+    heat: "heated",
+    time: null,
+    },{
+        output: {fluid: "kubejs:shimmer", amount: 1000 * mB },
+        input: [
+            "5x techreborn:amethyst_dust",
+            "2x minecraft:glow_ink_sac",
+            "ad_astra:moon_sand"
+        ],
+        heat: "superheated",
+        time: null,

```

Comment:
BTW: You can omit `time: null`, if code will try to read from a key that doesn't exist, it will return `undefined`.

Reply:
its there for easier possible rebalance